Sources/DVR/Session.swift

```diff
@@ -163,6 +163,25 @@ open class Session: URLSession {
         }
     }
 
+    func finishTask(_ task: URLSessionTask, responseData: Data?, playback: Bool) {
+        needsPersistence = (needsPersistence || !playback)
+
+        if let index = outstandingTasks.firstIndex(of: task) {
+            outstandingTasks.remove(at: index)
+        }
+
+        if !recording && outstandingTasks.count == 0 {
+            finishRecording()
+        }
+
+        if let delegate = delegate as? URLSessionDataDelegate, let task = task as? URLSessionDataTask, let data = responseData {
+            delegate.urlSession?(self, dataTask: task, didReceive: data as Data)
+        }
+
+        if let delegate = delegate as? URLSessionTaskDelegate {
+            delegate.urlSession?(self, task: task, didCompleteWithError: nil)
+        }
+    }
```

Comment:
I'm unsure why that's green. None of that is new code. In fact I didn't make any changes to that code at all.

Reply:
This is new code. Previously, when you did not want an interaction persisted it still created and passed an interaction as a shell for the filtered response data which was confusing because it was creating an interaction when you did not want it to (even though logically it was being thrown out). I wasn't sure if the logic would uphold with my changes so I added this to handle the filtered response data without adding creating the shell filter to ensure no unnecessary logic issues arise.

Reply:
It was pretty much copied from the existing function with the Interaction parts removed.

Reply:
It might make more sense then, rather than duplicating code, to modify the finish task code to accept an optional interaction. Otherwise if some logic changes in finishTask, it will need to be updated in two places, risking logic bugs.